external/kafka-0-10-sql/src/main/scala/org/apache/spark/sql/kafka010/KafkaContinuousReader.scala

```diff
@@ -164,7 +164,15 @@ case class KafkaContinuousDataReaderFactory(
     startOffset: Long,
     kafkaParams: ju.Map[String, Object],
     pollTimeoutMs: Long,
-    failOnDataLoss: Boolean) extends DataReaderFactory[UnsafeRow] {
+    failOnDataLoss: Boolean) extends ContinuousDataReaderFactory[UnsafeRow] {
+
+  override def createDataReaderWithOffset(offset: PartitionOffset): DataReader[UnsafeRow] = {
+    val kafkaOffset = offset.asInstanceOf[KafkaSourcePartitionOffset]
+    assert(kafkaOffset.topicPartition == topicPartition)
```

Comment:
This may happen. I prefer to use `require` like this:
```
require(kafkaOffset.topicPartition == topicPartition, s"expected: $topicPartition actual: ${kafkaOffset.topicPartition}")
```

Reply:
Got it.